### PR TITLE
Update the doc of GpuListUtils to eliminate a linkage warning.

### PR DESCRIPTION
**Pull request by firestarman**
This fixes https://github.com/NVIDIA/spark-rapids/issues/3520.

And the warning log on `GpuPythonUDF` is no longer found in recent builds.

Scaladoc will always try to create a link for the first word after the `@throws`, so here chooses to change it to a common comment instead of a tag.

Signed-off-by: Firestarman <firestarmanllc@gmail.com>

### CODE REVIEW

**Comment by firestarman** [comment]
build

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuListUtils.scala [hunk 1]
```diff
@@ -30,9 +30,8 @@ object GpuListUtils extends Arm {
    * both listCol and newDataCol alive longer than the returned ColumnView.
    * @param listCol the list column to use as a template
    * @param newDataCol the new data column.
-   * @return a new ColumnView.
-   * @throws scala.IllegalArgumentException if data column does not match the original data column
-   *                                        in size.
+   * @return a new ColumnView. But throws an scala.IllegalArgumentException if data column does
```
Comment (jlowe) [inline sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuListUtils.scala:33]: The only reason `scala.IllegalArgumentException` was originally used was to try to appease scaladoc.  We should just use IllegalArgumentException.
```suggestion
   * @return a new ColumnView. Throws IllegalArgumentException if data column does
```

**Review by jlowe — COMMENTED** [review]
It would be interesting to study how Apache Spark is setting up scaladoc, as they have `@throws IllegalArgumentException` directives in their scaladoc.  It looks like they too get linkage errors for this when building under Maven, but the instructions point to using `sbt unidoc` to generate their Scaladoc which may work around the linkage problems we're seeing here.

**Comment by firestarman** [comment]
build

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuListUtils.scala (hunk 1 above)
Comment (firestarman) [reply sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuListUtils.scala]: Updated

**Comment by firestarman** [comment]
> It would be interesting to study how Apache Spark is setting up scaladoc, as they have `@throws IllegalArgumentException` directives in their scaladoc. It looks like they too get linkage errors for this when building under Maven, but the instructions point to using `sbt unidoc` to generate their Scaladoc which may work around the linkage problems we're seeing here.

Yes, so it is not easy to clean all the warnings. You can find more such warnings in recent nightly builds besides this one. 
Here is [another fix](https://github.com/firestarman/spark-rapids/commit/2cf3ac4bb9d28e13e14417698ca8a2cbce8f7191) for this issue, to which I prefer. It should fix all the linkage warnings for the scala lang classes/objects.
What's your idea ?

**Comment by jlowe** [comment]
> Here is another fix for this issue, to which I prefer.

That approach seems fine if it fixes the issue.  Annoying that we have to do that extent of hand-holding for scaladoc, but I like that we don't have to avoid the `throws` directive for exceptions that are external to the module.

**Comment by firestarman** [comment]
This is replaced by https://github.com/NVIDIA/spark-rapids/pull/3673.

**Pull request closed without merging**